diff --git a/cinder/scheduler/filters/driver_filter.py b/cinder/scheduler/filters/driver_filter.py
index edfeb9f2d..42e1076a6 100644
--- a/cinder/scheduler/filters/driver_filter.py
+++ b/cinder/scheduler/filters/driver_filter.py
@@ -13,6 +13,8 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
+from keystoneauth1 import loading as ks_loading
+from oslo_config import cfg
 from oslo_log import log as logging
 
 from cinder.scheduler.evaluator import evaluator
@@ -20,6 +22,7 @@ from cinder.scheduler import filters
 
 
 LOG = logging.getLogger(__name__)
+CONF = cfg.CONF
 
 
 class DriverFilter(filters.BaseBackendFilter):
@@ -139,6 +142,19 @@ class DriverFilter(filters.BaseBackendFilter):
         request_spec = filter_properties.get('request_spec', {})
         volume_stats = request_spec.get('volume_properties', {})
 
+        LOG.debug('Creating Keystone auth plugin from conf')
+        n_auth = ks_loading.load_auth_from_conf_options(CONF, 'nova')
+        session = ks_loading.load_session_from_conf_options(
+            CONF, 'nova', auth=n_auth)
+        project_id = request_spec["volume"]["project_id"]
+        resp = session.get('/projects/%s' % project_id,
+                   endpoint_filter={'service_type': 'identity',
+                                    'interface': 'internal',
+                                    'min_version': '3',
+                                    'max_version': 'latest'})
+        data = resp.json()
+        volume_stats["domain_id"] = data["project"]["domain_id"]
+
         stats = {
             'backend_stats': backend_stats,
             'backend_caps': backend_caps,
