diff --git a/kolla/common/sources.py b/kolla/common/sources.py
index e7a72cd..62b89c6 100644
--- a/kolla/common/sources.py
+++ b/kolla/common/sources.py
@@ -65,7 +65,7 @@
                      'glance-${openstack_branch}.tar.gz')},
     'gnocchi-base': {
         'type': 'git',
-        'reference': '4.4.2',
+        'reference': 'stable/4.5',  # TODO(mmalchuk) change to >4.5.0 when released
         'location': ('https://github.com/gnocchixyz/'
                      'gnocchi.git')},
     'heat-base': {
diff --git a/kolla/image/unbuildable.py b/kolla/image/unbuildable.py
index 12f0e83..e90a9b3 100644
--- a/kolla/image/unbuildable.py
+++ b/kolla/image/unbuildable.py
@@ -17,7 +17,6 @@
 UNBUILDABLE_IMAGES = {
     'aarch64': {
         "bifrost-base",        # someone need to get upstream working first
-        "gnocchi-base",        # https://github.com/gnocchixyz/gnocchi/issues/1304 # noqa
         "prometheus-msteams",  # no aarch64 binary
         "prometheus-mtail",    # no aarch64 binary
         "skydive-base",        # no aarch64 binary
@@ -26,7 +25,6 @@
     # Issues for SHA1 keys:
     # https://github.com/grafana/grafana/issues/41036
     'centos': {
-        "gnocchi-base",          # https://github.com/gnocchixyz/gnocchi/issues/1304 # noqa
         "hacluster-pcs",         # Missing crmsh package
         "nova-spicehtml5proxy",  # Missing spicehtml5 package
         "ovsdpdk",               # Not supported on CentOS
@@ -34,11 +32,9 @@
     },
 
     'debian': {
-        "gnocchi-base",  # https://github.com/gnocchixyz/gnocchi/issues/1304
     },
 
     'rocky': {
-        "gnocchi-base",          # https://github.com/gnocchixyz/gnocchi/issues/1304 # noqa
         "hacluster-pcs",         # Missing crmsh package
         "nova-spicehtml5proxy",  # Missing spicehtml5 package
         "ovsdpdk",               # Not supported on CentOS
@@ -47,18 +43,15 @@
 
     'ubuntu': {
         "bifrost-base",  # Failing on new efi shim file location
-        "gnocchi-base",  # https://github.com/gnocchixyz/gnocchi/issues/1304
         "collectd",      # Missing collectd-core package
         "telegraf",      # Missing collectd-core package
     },
 
     'ubuntu+aarch64': {
         "barbican-base",  # https://github.com/unbit/uwsgi/issues/2434
-        "gnocchi-base",   # https://github.com/gnocchixyz/gnocchi/issues/1304
     },
 
     'centos+aarch64': {
-        "gnocchi-base",  # https://github.com/gnocchixyz/gnocchi/issues/1304
         "telegraf",      # no binary package
     },
 }
