--- a/kolla/image/build.py
+++ b/kolla/image/build.py
@@ -544,6 +544,12 @@ class BuildTask(DockerTask):

         self.logger.debug('Processing')

+        # NOTE(berendt): do not rebuild existing images, but allow the children to be built
+        if self.conf.skip_existing and image.in_docker_cache():
+            image.status = STATUS_BUILT
+            self.logger.info('Skipping %s, it already exists' % image.name)
+            return
+
         if image.status in [STATUS_SKIPPED, STATUS_UNBUILDABLE]:
             self.logger.info('Skipping %s' % image.name)
             return
