--- a/docker/elasticsearch/elasticsearch/Dockerfile.j2
+++ b/docker/elasticsearch/elasticsearch/Dockerfile.j2
@@ -21,7 +21,7 @@ ENV JAVA_HOME /usr/lib/jvm/jre-11-openjdk/

 {% elif base_package_type == 'deb' %}
     {% set elasticsearch_packages = [
-        'elasticsearch-oss',
+        'elasticsearch',
     ] %}

 # NOTE(osmanlicilegi): post-install script of elasticsearch fails when trying to

