diff --git a/kolla/image/kolla_worker.py b/kolla/image/kolla_worker.py
index b1e7ac943..639d729a8 100644
--- a/kolla/image/kolla_worker.py
+++ b/kolla/image/kolla_worker.py
@@ -322,7 +322,7 @@ class KollaWorker(object):
                           self.openstack_branch_slashed,
                       'tag': self.tag,
                       'maintainer': self.maintainer,
-                      'kolla_version': kolla_version,
+                      'kolla_version': '2024.2',
                       'image_name': image_name,
                       'users': self.get_users(),
                       'distro_python_version': self.distro_python_version,
