diff --git a/kolla/image/build.py b/kolla/image/build.py
index 1a2be084..9fc8fc33 100755
--- a/kolla/image/build.py
+++ b/kolla/image/build.py
@@ -766,7 +766,7 @@ class KollaWorker(object):
                 env.filters.update(self._get_filters())
                 env.globals.update(self._get_methods())
                 template = env.get_template(template_name)
-            content = template.render(values)
+            content = template.render(values, env=os.environ)
             content_path = os.path.join(path, 'Dockerfile')
             with open(content_path, 'w') as f:
                 LOG.debug("Rendered %s into:", tpl_path)
