diff --git a/kolla/common/config.py b/kolla/common/config.py
index ac98bb8f..20b403d1 100755
--- a/kolla/common/config.py
+++ b/kolla/common/config.py
@@ -205,6 +205,8 @@ _CLI_OPTS = [
                help='The Docker namespace name'),
     cfg.BoolOpt('cache', default=True,
                 help='Use the Docker cache when building'),
+    cfg.BoolOpt('squash', default=False,
+                help='Squash images'),
     cfg.MultiOpt('profile', types.String(), short='p',
                  help=('Build a pre-defined set of images, see [profiles]'
                        ' section in config. The default profiles are:'
diff --git a/kolla/image/build.py b/kolla/image/build.py
index 1a2be084..3f178abe 100755
--- a/kolla/image/build.py
+++ b/kolla/image/build.py
@@ -460,6 +460,7 @@ class BuildTask(DockerTask):
                                           rm=True,
                                           pull=pull,
                                           forcerm=self.forcerm,
+                                          squash=self.conf.squash,
                                           buildargs=buildargs):
                 stream = json.loads(response.decode('utf-8'))
                 if 'stream' in stream:
