--- a/doc/source/contributor/versions.rst
+++ b/doc/source/contributor/versions.rst
@@ -42,7 +42,7 @@ information about package sources.
 .. _`Kibana install guide`: https://www.elastic.co/guide/en/kibana/7.10/install.html
 .. _`Logstash install guide`: https://www.elastic.co/guide/en/logstash/7.9/installing-logstash.html
 .. _`TreasureData install guide`: https://www.fluentd.org/download
-.. _`ProxySQL repository`: https://repo.proxysql.com/ProxySQL/proxysql-2.7.x/
+.. _`ProxySQL repository`: https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/
 
 .. _`Team RabbitMQ 'Cloudsmith' repo (Deb)`: https://www.rabbitmq.com/install-debian.html#apt-cloudsmith
 .. _`Team RabbitMQ 'Modern Erlang' PPA`: https://launchpad.net/~rabbitmq/+archive/ubuntu/rabbitmq-erlang
--- a/docker/base/Dockerfile.j2
+++ b/docker/base/Dockerfile.j2
@@ -325,7 +325,7 @@ COPY apt_preferences /etc/apt/preferences.d/kolla-custom
    {'name': 'influxdb', 'url': 'https://repos.influxdata.com/influxdata-archive.key'},
    {'name': 'mariadb', 'url': 'https://downloads.mariadb.com/MariaDB/mariadb-keyring-2019.gpg', 'type': 'gpg'},
    {'name': 'opensearch', 'url': 'https://artifacts.opensearch.org/publickeys/opensearch.pgp'},
-   {'name': 'proxysql', 'url': 'https://repo.proxysql.com/ProxySQL/proxysql-2.7.x/repo_pub_key'},
+   {'name': 'proxysql', 'url': 'https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/repo_pub_key'},
    {'name': 'proxysql-3', 'url': 'https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/repo_pub_key'},
    {'name': 'treasuredata', 'url': 'https://packages.treasuredata.com/GPG-KEY-td-agent'},
 ] %}
--- a/docker/base/proxysql.repo
+++ b/docker/base/proxysql.repo
@@ -1,7 +1,7 @@
 [proxysql]
 name = ProxySQL 2.7.x
 # NOTE(mnasiadka): use 9 for both 9 and 10
-baseurl = https://repo.proxysql.com/ProxySQL/proxysql-2.7.x/almalinux/9
-gpgkey = https://repo.proxysql.com/ProxySQL/proxysql-2.7.x/repo_pub_key
+baseurl = https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/almalinux/$releasever
+gpgkey = https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/repo_pub_key
 gpgcheck = 1
 enabled = 0
--- a/kolla/template/repos.yaml
+++ b/kolla/template/repos.yaml
@@ -90,7 +90,7 @@ debian:
     component: "main"
     gpg_key: "opensearch.asc"
   proxysql:
-    url: "https://repo.proxysql.com/ProxySQL/proxysql-2.7.x/bookworm/"
+    url: "https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/bookworm/"
     suite: "./"
     component: ""
     gpg_key: "proxysql.asc"
@@ -152,7 +152,7 @@ debian-aarch64:
     component: "main"
     gpg_key: "opensearch.asc"
   proxysql:
-    url: "https://repo.proxysql.com/ProxySQL/proxysql-2.7.x/bookworm/"
+    url: "https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/bookworm/"
     suite: "./"
     component: ""
     gpg_key: "proxysql.asc"
@@ -259,7 +259,7 @@ ubuntu:
     component: "main"
     gpg_key: "opensearch.asc"
   proxysql:
-    url: "https://repo.proxysql.com/ProxySQL/proxysql-2.7.x/noble/"
+    url: "https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/noble/"
     suite: "./"
     component: ""
     gpg_key: "proxysql.asc"
@@ -322,7 +322,7 @@ ubuntu-aarch64:
     component: "main"
     gpg_key: "opensearch.asc"
   proxysql:
-    url: "https://repo.proxysql.com/ProxySQL/proxysql-2.7.x/noble/"
+    url: "https://repo.proxysql.com/ProxySQL/proxysql-3.0.x/noble/"
     suite: "./"
     component: ""
     gpg_key: "proxysql.asc"
