--- /var/lib/kolla/venv/lib/python3.11/site-packages/ironic/drivers/modules/redfish/bios.py	2025-08-15 10:57:01
+++ /var/lib/kolla/venv/lib/python3.11/site-packages/ironic/drivers/modules/redfish/bios.py	2025-08-15 10:57:07
@@ -29,7 +29,7 @@
 
 registry_fields = ('attribute_type', 'allowable_values', 'lower_bound',
                    'max_length', 'min_length', 'read_only',
-                   'reset_required', 'unique', 'upper_bound')
+                   'reset_required', 'unique')
 
 
 class RedfishBIOS(base.BIOSInterface):
