diff --git a/docker/openstack-base/Dockerfile.j2 b/docker/openstack-base/Dockerfile.j2
index a1120332e..f7cce1906 100644
--- a/docker/openstack-base/Dockerfile.j2
+++ b/docker/openstack-base/Dockerfile.j2
@@ -320,6 +320,7 @@ RUN ln -s openstack-base-source/* /requirements \
     && virtualenv --system-site-packages /var/lib/kolla/venv
 
 ENV PATH /var/lib/kolla/venv/bin:$PATH
+ENV SETUPTOOLS_USE_DISTUTILS=stdlib
 
 RUN {{ macros.install_pip(openstack_base_pip_packages | customizable("pip_packages")) }}
 
