diff --git a/docker/macros.j2 b/docker/macros.j2
index ddb1f351..041f8204 100644
--- a/docker/macros.j2
+++ b/docker/macros.j2
@@ -39,7 +39,8 @@ RUN usermod --append --home {{ homedir }} --groups kolla {{ name }} \
 {% macro ubuntu_trove_bug_1651852() %}
 {# NOTE(SamYaple): The postinst script breaks because it calls getenv instead of getent #}
 {# TODO(SamYaple): Remove once issue is fixed -- https://bugs.launchpad.net/ubuntu/+source/openstack-trove/+bug/1651852 #}
-RUN apt-get -y install --no-install-recommends trove-common \
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends trove-common \
     || sed -i 's/getenv/getent/g' /var/lib/dpkg/info/trove-common.postinst \
     && apt-get -y install -f \
     && apt-get clean
@@ -48,8 +49,9 @@ RUN apt-get -y install --no-install-recommends trove-common \
 
 {% macro debian_haproxy_existing_user_fix() %}
 {# NOTE(SamYaple): The postinst script breaks if the user 'haproxy' already exists #}
-RUN apt-get -y install --no-install-recommends haproxy \
-    || sed -i '/^adduser/,+1 d' /var/lib/dpkg/info/haproxy.postinst \
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends haproxy \
+    || ( test -e /var/lib/dpkg/info/haproxy.postinst && sed -i '/^adduser/,+1 d' /var/lib/dpkg/info/haproxy.postinst ) \
     && apt-get -y install -f \
     && apt-get clean
 {% endmacro %}
diff --git a/kolla/template/methods.py b/kolla/template/methods.py
index 8e655f4c..08c1280b 100644
--- a/kolla/template/methods.py
+++ b/kolla/template/methods.py
@@ -44,6 +44,7 @@ def debian_package_install(packages):
 
     # handle the apt-get install
     if reg_packages:
+        cmds.append('apt-get update')
         cmds.append('apt-get -y install --no-install-recommends {}'.format(
             ' '.join(reg_packages)
         ))
