diff --git a/kolla/image/build.py b/kolla/image/build.py
index 7fd6fd71..bae238f4 100755
--- a/kolla/image/build.py
+++ b/kolla/image/build.py
@@ -670,7 +670,7 @@ class KollaWorker(object):
                       'namespace': self.namespace,
                       'tag': self.tag,
                       'maintainer': self.maintainer,
-                      'kolla_version': kolla_version,
+                      'kolla_version': 'ocata',
                       'image_name': image_name,
                       'users': self.get_users(),
                       'rpm_setup': self.rpm_setup}
