diff --git a/docker/neutron/neutron-sfc-agent/Dockerfile.j2 b/docker/neutron/neutron-sfc-agent/Dockerfile.j2
deleted file mode 100644
index 4210483c..00000000
--- a/docker/neutron/neutron-sfc-agent/Dockerfile.j2
+++ /dev/null
@@ -1,15 +0,0 @@
-# TODO(egonzalez): SFC image is deprecated and will be removed during Queens release
-FROM {{ namespace }}/{{ image_prefix }}neutron-base:{{ tag }}
-LABEL maintainer="{{ maintainer }}" name="{{ image_name }}" build-date="{{ build_date }}"
-
-{% block neutron_networking_sfc_agent_header %}{% endblock %}
-
-{% import "macros.j2" as macros with context %}
-
-COPY extend_start.sh /usr/local/bin/kolla_neutron_extend_start
-RUN chmod 755 /usr/local/bin/kolla_neutron_extend_start
-
-{% block neutron_sfc_agent_footer %}{% endblock %}
-{% block footer %}{% endblock %}
-
-USER neutron
diff --git a/docker/neutron/neutron-sfc-agent/extend_start.sh b/docker/neutron/neutron-sfc-agent/extend_start.sh
deleted file mode 100644
index d721b4d7..00000000
--- a/docker/neutron/neutron-sfc-agent/extend_start.sh
+++ /dev/null
@@ -1,8 +0,0 @@
-#!/bin/bash
-
-# Bootstrap and exit if KOLLA_BOOTSTRAP variable is set. This catches all cases
-# of the KOLLA_BOOTSTRAP variable being set, including empty.
-if [[ "${!KOLLA_BOOTSTRAP[@]}" ]]; then
-    neutron-db-manage --subproject networking-sfc --config-file /etc/neutron/neutron.conf upgrade head
-    exit 0
-fi
-- 
2.13.1

