--- a/kolla/common/config.py
+++ b/kolla/common/config.py
@@ -568,10 +568,6 @@ SOURCES = {
         'type': 'url',
         'location': ('$tarballs_base/openstack/networking-sfc/'
                      'networking-sfc-stable-victoria.tar.gz')},
-    'neutron-base-plugin-vmware-nsx': {
-        'type': 'url',
-        'location': ('$tarballs_base/x/vmware-nsx/'
-                     'vmware-nsx-master.tar.gz')},
     'neutron-base-plugin-vpnaas-agent': {
         'type': 'url',
         'location': ('$tarballs_base/openstack/neutron-vpnaas/'
@@ -584,10 +580,6 @@ SOURCES = {
         'type': 'url',
         'location': ('$tarballs_base/openstack/neutron-dynamic-routing/'
                      'neutron-dynamic-routing-stable-victoria.tar.gz')},
-    'neutron-server-plugin-vmware-nsxlib': {
-        'type': 'url',
-        'location': ('$tarballs_base/x/vmware-nsxlib/'
-                     'vmware-nsxlib-master.tar.gz')},
     'neutron-vpnaas-agent': {
         'type': 'url',
         'location': ('$tarballs_base/openstack/neutron-vpnaas/'
