diff --git a/kolla/common/sources.py b/kolla/common/sources.py
index 42fa613b5..1755cb479 100644
--- a/kolla/common/sources.py
+++ b/kolla/common/sources.py
@@ -91,22 +91,10 @@ SOURCES = {
         'type': 'url',
         'location': ('$tarballs_base/openstack/horizon/'
                      'horizon-${openstack_branch}.tar.gz')},
-    'horizon-plugin-blazar-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/blazar-dashboard/'
-                     'blazar-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-cloudkitty-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/cloudkitty-dashboard/'
-                     'cloudkitty-dashboard-${openstack_branch}.tar.gz')},
     'horizon-plugin-designate-dashboard': {
         'type': 'url',
         'location': ('$tarballs_base/openstack/designate-dashboard/'
                      'designate-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-freezer-web-ui': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/freezer-web-ui/'
-                     'freezer-web-ui-${openstack_branch}.tar.gz')},
     'horizon-plugin-heat-dashboard': {
         'type': 'url',
         'location': ('$tarballs_base/openstack/heat-dashboard/'
@@ -127,14 +115,6 @@ SOURCES = {
         'type': 'url',
         'location': ('$tarballs_base/openstack/masakari-dashboard/'
                      'masakari-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-mistral-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/mistral-dashboard/'
-                     'mistral-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-murano-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/murano-dashboard/'
-                     'murano-dashboard-${openstack_branch}.tar.gz')},
     'horizon-plugin-neutron-vpnaas-dashboard': {
         'type': 'url',
         'location': ('$tarballs_base/openstack/neutron-vpnaas-dashboard/'
@@ -143,42 +123,6 @@ SOURCES = {
         'type': 'url',
         'location': ('$tarballs_base/openstack/octavia-dashboard/'
                      'octavia-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-sahara-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/sahara-dashboard/'
-                     'sahara-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-senlin-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/senlin-dashboard/'
-                     'senlin-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-solum-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/solum-dashboard/'
-                     'solum-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-tacker-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/tacker-horizon/'
-                     'tacker-horizon-${openstack_branch}.tar.gz')},
-    'horizon-plugin-trove-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/trove-dashboard/'
-                     'trove-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-venus-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/venus-dashboard/'
-                     'venus-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-vitrage-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/vitrage-dashboard/'
-                     'vitrage-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-watcher-dashboard': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/watcher-dashboard/'
-                     'watcher-dashboard-${openstack_branch}.tar.gz')},
-    'horizon-plugin-zun-ui': {
-        'type': 'url',
-        'location': ('$tarballs_base/openstack/zun-ui/'
-                     'zun-ui-${openstack_branch}.tar.gz')},
     'ironic-base': {
         'type': 'url',
         'location': ('$tarballs_base/openstack/ironic/'
